sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -150,10 +162,7 @@ export function abortSignal(signal: AbortSignal) {
     signal.onabort.call(signal);
   }
 
-  const listeners = listenersMap.get(signal)!;
-  listeners.forEach((listener) => {
-    listener.call(signal);
-  });
+  signal.dispatchEvent({ type: 'abort' })
```

Comment:
If `dispatchEvent` is updated to handle `onabort`, this function can be updated to remove the `onabort` check/invocation.